src/screenshot.cpp

```diff
@@ -856,3 +864,83 @@ bool MakeScreenshot(ScreenshotType t, const char *name)
 
 	return ret;
 }
+
+static byte _owner_colours[OWNER_END + 1];
+
+/**
+ * Return the colour a tile would be displayed with in the small map in mode "Owner".
+ *
+ * @param tile The tile of which we would like to get the colour.
+ * @return The colour of tile in the small map in mode "Owner"
+ */
+static inline byte GetMinimapPixels(TileIndex tile)
+{
+	Owner o;
+
+	switch (GetTileType(tile)) {
+	case MP_INDUSTRY: o = OWNER_END;          break;
+	case MP_HOUSE:    o = OWNER_TOWN;         break;
+	default:          o = GetTileOwner(tile); break;
+		/* FIXME: For MP_ROAD there are multiple owners.
+		 * GetTileOwner returns the rail owner (level crossing) resp. the owner of ROADTYPE_ROAD (normal road),
+		 * even if there are no ROADTYPE_ROAD bits on the tile.
+		 */
+	}
+
+	return _owner_colours[o];
+}
+
+static void MinimapScreenCallback(void *userdata, void *buf, uint y, uint pitch, uint n)
+{
+	uint32 *ubuf = (uint32 *)buf;
+
+	uint num = (pitch * n);
+	uint row, col;
+	byte val;
+
+	for (uint i = 0; i < num; i++) {
+		row = y + (int)(i / pitch);
+		col = (MapSizeX() - 1) - (i % pitch);
+
+		TileIndex tile = TileXY(col, row);
+
+		if (IsTileType(tile, MP_VOID)) {
+			val = 0x00;
+		}
+		else
+		{
+			val = GetMinimapPixels(tile);
+		}
+
+		uint32 colour_buf = 0;
+		colour_buf  = (_cur_palette.palette[val].b << (sizeof(uint8) * 0));
+		colour_buf |= (_cur_palette.palette[val].g << (sizeof(uint8) * 8));
+		colour_buf |= (_cur_palette.palette[val].r << (sizeof(uint8) * 16));
+		*ubuf = colour_buf;
+		 ubuf += sizeof(uint8);   // Skip alpha
+	}
+}
+
+/**
+ * Make a minimap screenshot.
+ * @param name the name to give to the screenshot.
+ */
+bool MakeMinimapWorldScreenshot()
+{
+	/* setup owner table */
+	const Company *c;
```

Comment:
No need to declare this so early. Put it close to first use (since you can't declare it in the FOR_ALL_COMPANIES block). Also be good if you can capitalise your comments correctly